lua/indentmini/init.lua

```diff
@@ -46,11 +42,19 @@ local function indentline()
       indent = prev > 20 and 4 or prev
     end
 
-    local hi_name = hl_group()
-
     ctx[row] = indent
 
-    for i = 1, indent - 1, indent_step(bufnr) do
+    local shiftw = indent_step(bufnr)
+    for i = 1, indent - 1, shiftw do
+      local hi_name = 'IndentLine'
+      if opts.hi_colors and #opts.hi_colors ~= 0 then
```

Comment:
thanks for this . i would say it's nice. and i think we can simply handle this just set the indent highlight group to indengline + level and default link to `IndentLine` if someone want colorful just config it by `nvim_set_hl` or hi command . so we don't need a config param in setup.

Reply:
Okay I've made some updates based on your suggestion. Now you just have to create highlight groups for `IndentLineX`. You can add as many or few as you want, and extend how it works pretty easily.

I had another solution that was more restrictive, I think indent-blankline used it in previous version. Where you have a list of allowed highlight groups that can be used. But this is definitely cleaner.

I'm still working out some kinks and edge-cases with it, but let me know if this is the type of usage you were thinking of.

Reply:
Out of curiosity, why do you want to avoid adding a config param for something like this? I want to understand so I can help better.